judge/views/problem.py

```diff
@@ -335,7 +336,7 @@ def profile(self):
         return self.request.profile
 
     def get_normal_queryset(self):
-        filter = Q(is_public=True)
+        filter = Q(is_public=True, is_suggesting=False)
```

Comment:
We don't need `is_suggesting=False` because if a problem is in the suggesting state, it should not be public